[WIP] Replace Sidetiq w/ Sidekiq-Cron

chrisarcand opened the pull request:
Closes #282

This is better besides, but became more important when Sidetiq closed all of its issues and wiki, rendering most resources on how to use it completely worthless 😠 

Drops celluloid dependency, too.

Fryguy commented:
This is cool.  A few comments:

- Can you add a comment to the .yml file that's human readable, so I don't have to parse cron in my head to understand it?  For example:

```yaml
batch_job_monitor:
  # minutely
  cron: '* * * * *'
  class: 'BatchJobMonitor'
  queue: 'miq_bot'

commit_monitor:
  # every 5 minutes
  cron: '*/5 * * * *'
  class: 'CommitMonitor'
  queue: 'miq_bot_glacial'
```

chrisarcand commented:
Added human readable comments and removed queue selection from the schedule as the workers can specify that (if I could put all sidekiq options in the schedule I'd do that but...alas, gem doesn't allow you to do that, so can't add retry boolean)

> The initializer checks if Sidekiq.server?, but we run multiple sidekiq servers.

Eh...I don't know. Unfamiliar with how we do sidekiq then...might just have to try it.

chrisarcand commented:
Regarding the two processes, I think we're good. Here's the check: https://github.com/mperham/sidekiq/blob/2a07a551aa8bc2356bdec736cc349c433d1408ca/lib/sidekiq.rb#L86-L88

I think the crons are only registered once. And I've tested locally with foreman running our two queues separately and it seems fine.

bdunne commented on config/initializers/sidekiq.rb:
```diff
@@ -0,0 +1,5 @@
+schedule_file = "config/worker_schedule.yml"
+
+if File.exist?(schedule_file) && Sidekiq.server?
+  Sidekiq::Cron::Job.load_from_hash YAML.load_file(schedule_file)
```
Missing parens `Sidekiq::Cron::Job.load_from_hash(YAML.load_file(schedule_file))`

Fryguy replied:
Why didn't rubocop flag this?

Fryguy commented:
The workers I'd be concerned about already check for uniqueness, so even if it does spawn multiple, I don't think it will hurt.  I think we can deploy and then watch the logs to verify

chrisarcand commented:
@bdunne @Fryguy Let me dig just a bit further, then.

chrisarcand commented:
Ugh you're right. It's loading twice. I'll have to do it inline.

chrisarcand commented:
@miq-bot add_label wip

chrisarcand commented:
Yeah this just isn't going to work without some extra elbow grease. I've updated the issue with my findings.
